src/distances/delta.jl

```diff
@@ -1,11 +1,12 @@
 struct Delta <: Distances.PreMetric
 end
 
-@inline function Distances._evaluate(::Delta, a::AbstractVector, b::AbstractVector) where {T}
+@inline function Distances._evaluate(::Delta, a::AbstractVector{Ta}, b::AbstractVector{Tb}) where {Ta, Tb}
     @boundscheck if length(a) != length(b)
         throw(DimensionMismatch("first array has length $(length(a)) which does not match the length of the second, $(length(b))."))
     end
-    return a == b
+    T = promote_type(Ta, Tb)
+    return T(a == b)
```

Comment:
This is amazingly picky, but I think it's probably more clear to write
```suggestion
    return convert(promote_type(Ta, Tb), a == b)
```

Reply:
Actually, I don't think this implementation will work properly with `ColVecs`. @sharanry could you test this please before we merge this PR?

Reply:
Oh yeah you're right! I just thought of the simplest case. Well it actually leads to the issue to what the output type of `Delta` should be...

Reply:
I mean, I think it's fine for it to be a `Bool`. I can't see what harm that causes -- when it interacts with a variance parameter that is e.g. a `Float32` it should be appropriately promoted.